cmake/modules/SearchInstalledSoftware.cmake

```
@@ -552,7 +552,13 @@ find_package(Python3 3.8 COMPONENTS ${python_components})
 #---Check for OpenGL installation-------------------------------------------------------
 if(opengl)
   message(STATUS "Looking for OpenGL")
-  find_package(OpenGL)
+  if(APPLE)
+    set(CMAKE_FIND_FRAMEWORK FIRST)
+    find_package(OpenGL)
+    set(CMAKE_FIND_FRAMEWORK LAST)
+  else()
+    find_package(OpenGL)
+  endif()
```

```suggestion
  set(CMAKE_FIND_FRAMEWORK FIRST)
  find_package(OpenGL)
  set(CMAKE_FIND_FRAMEWORK LAST)
```
According to the [cmake documentation](https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_FRAMEWORK.html), this flag only has an effect on macOS anyway. So what is the benefit of having an extra `if(APPLE)` branch?

---

It s Turner to LAST b4

---

Correct, and it is not inside a `if(APPLE)` branch. My question was why we need the `if(APPLE)` now? Sorry for the misunderstanding

---

Sorry, maybe I was not clear enough to start with. The `CMAKE_FIND_FRAMEWORK` flag was set to FALSE for all machines years ago, which means that system libraries on all platforms are checked "last". However, on macOS, we now need the system libraries to be checked `FIRST` for OpenGL. This is specific to macOS because the OpenGL component of ROOT requires the system's version of OpenGL to function correctly. Therefore, on macOS, just before locating OpenGL, we change the flag to `FIRST`, and immediately after, we switch it back to `LAST` to maintain the same behavior as before on other platforms.